packages/core/src/tracing/instrumentation/database/elasticsearchModern.js

```diff
@@ -195,12 +215,17 @@ function parseConnectionFromResult(span, result) {
       span.data.elasticsearch.address = connectionUrl.hostname;
       span.data.elasticsearch.port = connectionUrl.port;
     }
+  } else if (connectionString) {
+    const parsedConnectionUrl = new URL(connectionString);
+    span.data.elasticsearch.address = parsedConnectionUrl.hostname;
+    span.data.elasticsearch.port = parsedConnectionUrl.port;
```

Comment:
_suggestion:_ We could potentially shave off some CPU cycles by moving the `new URL(...)` into the monkey-patched `es.Client = function InstanaClient() {`  function I think. If I am not mistaken, the way it is implemented now, we parse the URL on every request, although it is always the same `connectionString` value.

OTOH, users might instantiate multiple instances of `es.Client`, talking to different ElasticSearch clusters/servers, I guess? Could keeping only one `connectionString` might lead to scenarios where we record the wrong target address/port for some calls?

Reply:
I will put the URL parsing to the patched client override for now.

> OTOH, users might instantiate multiple instances of es.Client

Yes totally true. But I think there are a lot of instrumentations which ignore this use case and store data in global variables inside the instrumentations. I'd like to ignore this case.